core/src/main/scala/org/apache/spark/util/collection/Utils.scala

```diff
@@ -79,6 +79,20 @@ private[spark] object Utils {
     builder.result()
   }
 
+  /**
+   * Same function as `keys.zipWithIndex.toMap`, but has perf gain.
+   */
+  def toMap[K](keys: Iterable[K]): Map[K, Int] = {
```

Comment:
`toMapWithIndex`?

Reply:
ok

Reply:
[405c625](https://github.com/apache/spark/pull/37940/commits/405c6252893c0df6cdf57cea8da94d1ebe7f56c4) fix this, waiting ci

Reply:
GA passed